### ISSUE TITLE
zfp python wrapper problems with "big" multidimensional arrays

### ISSUE THREAD

**Issue by romankarlstetter** [LLNL/zfp#45]
I experience a problem when compressing some 3d numpy array (using the implementation discussed in #29, using ebe63fa16213eb8d5cb09068e2f8f973d0d694f8 from branch feature/py-bindings).

When compressing an array that has a shape of `(4, 4, s)`, where `s > 2^16`, the shape of the decompressed array is not correct anymore.
### How to reproduce:
```python
shp = (4, 4, 65537)
ar = np.zeros(shp)
_comprsd_3d = zfp.compress_numpy(ar, tolerance=0.2)
decompressed = zfp.decompress_numpy(_comprsd_3d)
print(decompressed.shape) # --> prints (4, 4, 1)
```

When using a shape of `(16, 65537)`, this does not happen, so it does not seem to be related to the total size of the array.

### Expected
Shape of numpy array is the same after decompression for any size. Or, if that is not possible, some kind of error during compression (or at least decompression).

**Comment by lindstro** [comment]
This is currently a limitation of zfp's very concise header, which limits each array to 2^48 elements.  In d dimensions, each dimension is at most 2^(48/d) elements, i.e., 2^16 in 3D.  I agree that this is not very satisfactory, and is something that we will address in the future.  For now, this should be documented and we should throw an exception when dimensions are out of range.

**Comment by romankarlstetter** [comment]
Maybe this limitation should also be mentioned in https://zfp.readthedocs.io/en/release0.5.4/limitations.html

**Comment by lindstro** [comment]
@romankarlstetter I've pushed a "fix" on the develop branch that now checks the array dimensions and makes sure they can be encoded in the header.  If not, then `zfp_write_header` fails, which will also cause the Python compression code to report failure.  The documentation has been updated to include this limitation.  We plan on including this fix in the next release.

This isn't the most satisfactory solution, but to allow larger arrays we would have to change the header format, which would raise compatibility issues.  We'll provide a better solution with the next revision of the zfp format.
